File: Misc/100_lc_1343.java
package Misc;

/*
 * 1343. Number of Sub-arrays of Size K and Average Greater than or Equal to Threshold
 */

 class Solution {
    public int numOfSubarrays(int[] arr, int k, int threshold) {
        double currentAverage = 0;
        int currentSum = 0;
        for (int i=0; i<k; i++) {
            currentSum += arr[i];
        }

        currentAverage = currentSum / k;
        int startSWIndex = 1;
        int endSWIndex = k;
        int res = 0;
        if (currentAverage >= threshold) {res++;}

        while (endSWIndex < arr.length) {
            currentAverage = currentAverage + (((double) arr[endSWIndex] - (double) arr[startSWIndex-1]) / k);
            if (currentAverage >= threshold) {
                res++;
            }
            endSWIndex++;
            startSWIndex++;
        }
        return res;
    }
}

File: Misc/100_lc_1343.py
"""
1343. Number of Sub-arrays of Size K and Average Greater than or Equal to Threshold
"""

from typing import List


class Solution:
    def numOfSubarrays(self, arr: List[int], k: int, threshold: int) -> int:
        currentAverage = sum(arr[:k]) / k
        startSWIndex = 1
        endSWIndex = k
        res = 0
        if currentAverage >= threshold:
            res += 1

        while endSWIndex < len(arr):
            currentAverage = (
                currentAverage + (arr[endSWIndex] - arr[startSWIndex - 1]) / k
            )
            if currentAverage >= threshold:
                res += 1

            startSWIndex += 1
            endSWIndex += 1

        return res
